aap_chatbot/src/App.test.tsx

```diff
@@ -822,3 +822,22 @@ test("Chat streaming error in status check", async () => {
   const textContent = alert?.textContent;
   expect(textContent).not.contain("mocked error");
 });
+
+test("Chat service returns 429 Too Many Requests error in streaming", async () => {
+  mockAxios(200, false, false, referencedDocumentExample, true);
```

Comment:
I'm confused :smile: 

The _streaming_ version of the Chatbot seems to use [`fetchEventSource(..)`](https://github.com/ansible/ansible-ai-connect-service/blob/4996c458cfb50f87256378ce217fa20c4dda8076/aap_chatbot/src/useChatbot/useChatbot.ts#L443) whereas this test is using `mockAxiox(....., true)`. I'd have expected this test to mocked [`mockFetchEventSource()`](https://github.com/ansible/ansible-ai-connect-service/blob/4996c458cfb50f87256378ce217fa20c4dda8076/aap_chatbot/src/App.test.tsx#L336C5-L336C45).

The code looks fine.. I'm just confused as to why you mock one thing; but the code appears to use something different.

Reply:
I admit it's very confusing. 

It is for mocking [this health check call](https://github.com/ansible/ansible-ai-connect-service/blob/main/aap_chatbot/src/useChatbot/useChatbot.ts#L175).  The communication receiving chatbot data is mocked with `mockFetchEventSource()`.